lib/sheet.js

```diff
@@ -0,0 +1,76 @@
+import Spreadsheet from 'edit-google-spreadsheet';
+import googleAuth from './google-auth';
+import util from './util';
+
+let SHEET_CONFIG;
+util.readJsonFromFile('./google-spreadsheet-config.json', data => {
+    SHEET_CONFIG = data;
+});
+
+class Sheet {
+    constructor() {
+        this.dataCache = {};
+        return this;
+    }
+
+    fetch(searchString, processingCallback, afterRequestCallback){
+        if (!this.dataCache.extensions) {
+            this.updateCache(searchString, processingCallback, afterRequestCallback);
+        }else{
+            processingCallback(searchString, this.dataCache, afterRequestCallback);
+        }
+    }
+
+    updateCache(searchString, processingCallback, afterRequestCallback){
+        googleAuth.authenticate(tokens => {
+            this.requestSpreadsheet(SHEET_CONFIG.extensionsWorksheetId, SHEET_CONFIG.spreadsheetId, tokens, (extensions) => {
+                this.requestSpreadsheet(SHEET_CONFIG.aliasesWorksheetId, SHEET_CONFIG.spreadsheetId, tokens, (aliases) => {
+                    this.dataCache = this.buildData(extensions, aliases);
+                    if (processingCallback){
+                        processingCallback(searchString, this.dataCache, afterRequestCallback);
+                    }
+                });
+            });
+        });
+    }
+
+    requestSpreadsheet(worksheetId, spreadsheetId, tokens, callback){
+        Spreadsheet.load({
+            debug: false, //SWITCH TRUE/FALSE
+            spreadsheetId: spreadsheetId,
+            worksheetId: worksheetId,
+            accessToken: {
+                type: tokens.token_type,
+                token: tokens.access_token
+            }
+        }, (err, data) => {
+            if (err) {
+                throw err;
+            }
+            data.receive((err, rows, info) => {
+                if (err) {
+                    throw err;
+                }
+                callback(rows);
+            });
+        });
+    }
+
+    buildData(extensions, aliases){
+        let data = {
+            'extensions': {},
+            'aliases': {}
+        };
+
+        for (var i in extensions)
```

Comment:
Let us always use curly braces, even for single line stuff. See if you find value in these answers:
- http://stackoverflow.com/a/4797368/2389475
- http://stackoverflow.com/a/361327/2389475
- http://stackoverflow.com/a/12193202/2389475